### ISSUE TITLE
Form Select Component Validation

### ISSUE THREAD

**Issue by mehmetuken** [ant-design-blazor/ant-design-blazor#1036]
Select Component validation is not working. Any suggestions?

**Comment by anddrzejb** [comment]
Can you provide a simplified code repro? I am doing some work on form validation in PR  #1035, maybe I will be able to fix your problem.

**Comment by mehmetuken** [comment]
MyModel
```cs
public class CreateUpdateProductDto{
   [Required] public Guid CategoryId { get; set; }
}
```

NewEntity = CreateUpdateProductDto

FormItem binding value
```cs
<FormItem Label="Category">
                    <Select DataSource="@_categoryList"
                            PopupContainerSelector="#create-modal"
                            @bind-Value="@NewEntity.CategoryId"
                            LabelName="@nameof(CategoryLookupDto.Name)"
                            ValueName="@nameof(CategoryLookupDto.Id)"
                            Placeholder="Select Category"
                            DefaultActiveFirstItem="false"
                            OnSelectedItemChanged="@(async item => await OnCategorySelectedItemChanged(item))"
                            EnableSearch>
                    </Select>
</FormItem>
```

i use form ref validate method. It's show validate required dot but send empty not show validate message validate method send true.

**Comment by anddrzejb** [comment]
So...I think I found the problem, but I'd like you next time to provide me with fully re-creatable code (something that I can copy-paste and that is it). In this case I had to figure out your model, guess the content of the `OnSelectItemChanged`, guess missing form code, missing modal code, etc... This caused me to take more time to find out your problem.

And the problem is the fact that you are using `Guid`. It is not empty. Ever. So it does not catch it.
A simple fix is to use this model definition:
```csharp
    public class CreateUpdateProductDto
    {
        [Required]        
        [RegularExpression("^((?!00000000-0000-0000-0000-000000000000).)*$", ErrorMessage = "Cannot use default Guid")]
        public Guid CategoryId { get; set; }
    }
```
It will catch guid that has a default value (`00000000-0000-0000-0000-000000000000`).
Although I would probably just create another `ValidationAttribute` and use it like this:  
```csharp
        [Required]        
        [NotDefaultGuid]
        public Guid CategoryId { get; set; }
```
